library/Icingadb/Widget/Detail/UserDetail.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Widget\Detail;
+
+use Icinga\Module\Icingadb\Common\Auth;
+use Icinga\Module\Icingadb\Model\User;
+use Icinga\Module\Icingadb\Widget\EmptyState;
+use Icinga\Module\Icingadb\Widget\ItemList\UsergroupList;
+use ipl\Html\Attributes;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\Text;
+
+class UserDetail extends BaseHtmlElement
+{
+    use Auth;
+
+    /** @var User The given user */
+    protected $user;
+
+    protected $defaultAttributes = ['class' => 'user-detail'];
+
+    protected $tag = 'div';
+
+    public function __construct($user)
+    {
+        $this->user = $user;
+    }
+
+    protected function createCustomVars()
+    {
+        $content = [new HtmlElement('h2', null, Text::create(t('Custom Variables')))];
+        $flattenedVars = $this->user->customvar_flat;
+        $this->applyRestrictions($flattenedVars);
+
+        $vars = $this->user->customvar_flat->getModel()->unflattenVars($flattenedVars);
+        if (! empty($vars)) {
+            $customvarTable = new CustomVarTable($vars);
+            $customvarTable->setAttribute('id', 'user-customvars');
+            $content[] = $customvarTable;
+        } else {
+            $content[] = new EmptyState(t('No custom variables configured.'));
+        }
+
+        return $content;
+    }
+
+    protected function createUserDetail()
```

Comment:
* Why the parentheses around the constructors?
* Please use a `HorizontalKeyValue` widget for each key-value-pair, the `.key-value-list` isn't used anywhere else now and actually obsolete.